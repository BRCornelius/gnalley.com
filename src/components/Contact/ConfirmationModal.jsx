import React, { useRef } from 'react';
import { useOutsideClick } from '../../utils';
import './Contact.css';

export const ConfirmationModal = ({toggleFunction}) => {
    const wrapperRef = useRef(null)
    useOutsideClick(wrapperRef, toggleFunction)

    return <div className="modal-container">
        <h3>Thank you for contacting us. A qualified team member will get back to you as soon as we have a chance.</h3>
        <button onClick={toggleFunction} className="modal-close">Close</button>
    </div>
}