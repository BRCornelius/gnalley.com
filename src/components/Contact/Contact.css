.active {
    background-color: chartreuse;
    border-radius: 0.5rem;
    padding: 0.5rem;
}
.inactive {
    background-color: cornsilk;
    border-radius: 0.5rem;
    padding: 0.5rem;
}

h3 {
    margin: 0;
}

input {
    line-height: 1.5rem;
}

label {
    font-size: 1.25rem;
}

textarea {
    font-size: 1rem;
    line-height: 2rem;
    height: 15rem;
}

button {
    font-size: 1.25rem;
}

.button-container {
    display: inline-flex;
    justify-content: space-evenly;
    padding-left: 0.5rem;
    width: 50%;
}

.button-label {
    width: 50%;
}

.column {
    display: flex;
    flex-direction: column;
    width: 50%;
}

.container-row {
    align-items: center;
    display: flex;
    flex-direction: row;
    height: 100%;
    width: 100%;
}

.input-container {
    display: flex;
    flex-direction: column;
    padding-top: 2rem;
    padding-bottom: 2rem;
    padding-right: 2rem;
    width: 100%;
}

.radio-input-container {
    display: inline-flex;
    padding-top: 0.5rem;
}

.two-input-container {
    display: inline-flex;
}

.overarch-contact-container {
    background-color: #000035;
    border: 1px solid #ffffff;
    color: #ffffff;
    display: flex;
    flex-direction: column;
    padding: 0.5rem;
    width: -webkit-fill-available;
}

.input-message-container {
    display: flex;
    flex-direction: column;
    padding: 0.5rem;
}

.submit-button {
    background-color: chartreuse;
    border-radius: 1rem;
    font-weight: 800;
    padding: 1rem;
}

.submit-button-container {
    display: inline-flex;
    justify-content: center;
}

.modal-container {
    background-color: #ffffff;
    color: #003366;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    padding: 3rem;
    max-width: 100%;
    height: 100%;
}

.modal-close {
    margin-top: 1rem;
    padding: 1rem;
    border: 2px solid black;
}