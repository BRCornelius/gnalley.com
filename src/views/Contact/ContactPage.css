.two-columns {
    display: inline-flex;
    width: 100%;
}

.vertical-column {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 50%;
}

.vertical-column-right {
    border-left: 1px solid #f3e42e;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 50%;
}

.contact-page-container {
    height: 84vh;
  }