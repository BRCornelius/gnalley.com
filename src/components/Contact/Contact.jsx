import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import { GlobalContext } from '../../context/global-context';
import { isMobile } from '../../utils';
import { ConfirmationModal } from './ConfirmationModal';
import './Contact.css';

export const InputContainer = ({children}) => <div className={isMobile ? "mobile-input-container" : "input-container"} data-testid="input-field">{children}</div>

export const TextInput = ({label, stateValue, clickFunction, id}) => <InputContainer>
        <label>{label}</label>
        <input type="text" id={id} name={id} value={stateValue} onChange={event => clickFunction(event.target.value)}/>
    </InputContainer>
export const RadioInput = ({checkSelection, setSelection}) => <div className="radio-input-container" data-testid="input-field">
        <label className="button-label">Are you a new client?</label>
        <div className="button-container">
            <button className={checkSelection("yes") ? "active" : "inactive"} onClick={() => setSelection("yes")}>Yes</button>
            <button className={checkSelection("no") ? "active" : "inactive"} onClick={() => setSelection("no")}>No</button>
        </div>
    </div>
export const MessageInput = ({label, stateValue, clickFunction, id}) => <div className="input-message-container" data-testid="input-field">
        <label>{label}</label>
        <textarea type="text" id={id} value={stateValue} onChange={event => clickFunction(event.target.value)}/>
    </div>

export const Contact = ({isWidget}) => {
    let {
        email, firstName, lastName, message, newClient, open, phone, setEmail, sendEmail,
        setFirstName, setLastName, setMessage, setNewClient, setOpen, setPhone, emptyContactState
      } = useContext(GlobalContext)
    const toggleConfirmationModal = () => setOpen(!open)
    const checkNewClient = targetValue => targetValue === newClient;
    const history = useHistory();
    const handleNavigation = () => {
        history.push("/contact");
    }
    const handleSubmit = () => {
        toggleConfirmationModal();
        sendEmail();
        emptyContactState();
    }

    let header = isWidget
        ? <h3>Contact Us!<br/><small>Send us an email and we will get back to you as soon as possible.</small></h3>
        : <h1>Email us:</h1>

    const SubmitButton = () => <div className="submit-button-container">
        <button className="submit-button" onClick={handleSubmit}>Submit</button>
    </div>

    return <div className="overarch-contact-container">
        {!open && <>
            {header}
            {!isMobile && <div className="container-row">
                <div className="column">
                    <div className="two-input-container">
                        <TextInput label="First Name" stateValue={firstName} clickFunction={setFirstName} id="first-name" />
                        <TextInput label="Last Name" stateValue={lastName} clickFunction={setLastName} id="last-name" />
                    </div>
                    <div className="two-input-container">
                        <TextInput label="Email" stateValue={email} clickFunction={setEmail} id="email" />
                        <TextInput label="Phone Number" stateValue={phone} clickFunction={setPhone} id="phone" />
                    </div>
                    <RadioInput checkSelection={checkNewClient} setSelection={setNewClient} />
                </div>
                <div className="column">
                    <MessageInput label="Message" stateValue={message} clickFunction={setMessage} id="message" />
                    {isWidget && <SubmitButton />}
                </div>
            </div>}
            {isMobile && <>
                <TextInput label="First Name" stateValue={firstName} clickFunction={setFirstName} id="first-name" />
                <TextInput label="Last Name" stateValue={lastName} clickFunction={setLastName} id="last-name" />
                <TextInput label="Email" stateValue={email} clickFunction={setEmail} id="email" />
                <TextInput label="Phone Number" stateValue={phone} clickFunction={setPhone} id="phone" />
                <RadioInput checkSelection={checkNewClient} setSelection={setNewClient} />
                <MessageInput label="Message" stateValue={message} clickFunction={setMessage} id="message" />
                <SubmitButton />
            </>}
            {isWidget && <h3 className="more-info-text"><button onClick={handleNavigation}>Click Here</button> for other ways to contact us.</h3>}
            {!isWidget && !isMobile && <SubmitButton />}
        </>}
        {open && <ConfirmationModal toggleFunction={toggleConfirmationModal} /> }
    </div>
}