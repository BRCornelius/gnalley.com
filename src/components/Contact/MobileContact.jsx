import React from 'react';
import { InputContainer } from './Contact';
import './Contact.css';

export const MobileContact = () => <div className="mobile-contact-container">
        <h1>Email Us</h1>
        <InputContainer>
            
        </InputContainer>
    </div>