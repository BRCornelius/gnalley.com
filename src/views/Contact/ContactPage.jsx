import React from 'react';
import { Contact } from '../../components';
import './ContactPage.css';

export const ContactPage = () => <>
    <Contact isWidget={false}/>
    <div className="two-columns">
        <div className="vertical-column">
            <h1 className="contact-title">Call us:</h1>
            <h3>Phone: (504) 838-8188<br/>Fax: (504) 838-0008</h3>
        </div>
        <div className="vertical-column">
            <h1 className="contact-title">Our office:</h1>
            <h3>2450 Severn Avenue<br/>Suite 100<br/>Metairie, LA 70001</h3>
        </div>
    </div>
</>;