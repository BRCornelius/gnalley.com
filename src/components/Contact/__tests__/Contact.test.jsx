import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { GlobalContext } from '../../../context/global-context';
import { Contact } from '../Contact';

const renderWithContext = (contextObject) => render(<GlobalContext.Provider value={contextObject}><Contact /></GlobalContext.Provider>)

test('It should render properly', () => {
    const { container, getAllByTestId } = render(<Contact />);
    expect(getAllByTestId('input-field')).toHaveLength(6);
    expect(container).toMatchSnapshot()
})

test('It should change new client button when clicked', () => {
    const mockFunction = jest.fn();
    const providerProps = {
        email: '',
        firstNaem: '',
        lastName: '',
        message: '',
        newClient: null,
        open: false,
        phone: '',
        setEmail: mockFunction,
        sendEmail: mockFunction,
        setFirstName: mockFunction,
        setNewClient: jest.fn(() => providerProps.newClient = 'yes'),
        setLastName: mockFunction, setMessage: mockFunction, setOpen: mockFunction, setPhone: mockFunction, emptyContactState: mockFunction
    };
    const {getByText} = renderWithContext(providerProps)
    fireEvent.click(getByText('Yes'))
    expect(providerProps.setNewClient).toHaveBeenCalled();
    expect(providerProps.newClient === 'yes').toBeTruthy();
})