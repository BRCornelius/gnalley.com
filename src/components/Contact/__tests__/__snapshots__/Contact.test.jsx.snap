// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render properly 1`] = `
<div>
  <div
    class="overarch-contact-container"
  >
    <h1>
      Email us:
    </h1>
    <div
      class="mobile-input-container"
      data-testid="input-field"
    >
      <label>
        First Name
      </label>
      <input
        id="first-name"
        name="first-name"
        type="text"
        value=""
      />
    </div>
    <div
      class="mobile-input-container"
      data-testid="input-field"
    >
      <label>
        Last Name
      </label>
      <input
        id="last-name"
        name="last-name"
        type="text"
        value=""
      />
    </div>
    <div
      class="mobile-input-container"
      data-testid="input-field"
    >
      <label>
        Email
      </label>
      <input
        id="email"
        name="email"
        type="text"
        value=""
      />
    </div>
    <div
      class="mobile-input-container"
      data-testid="input-field"
    >
      <label>
        Phone Number
      </label>
      <input
        id="phone"
        name="phone"
        type="text"
        value=""
      />
    </div>
    <div
      class="radio-input-container"
      data-testid="input-field"
    >
      <label
        class="button-label"
      >
        Are you a new client?
      </label>
      <div
        class="button-container"
      >
        <button
          class="inactive"
        >
          Yes
        </button>
        <button
          class="inactive"
        >
          No
        </button>
      </div>
    </div>
    <div
      class="input-message-container"
      data-testid="input-field"
    >
      <label>
        Message
      </label>
      <textarea
        id="message"
        type="text"
      />
    </div>
    <div
      class="submit-button-container"
    >
      <button
        class="submit-button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
